Include/Library/OcCryptoLib.h

```diff
@@ -70,19 +68,77 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 #endif
 
 //
-// For now abort on anything but 2048, but we can support 1024 and 4096 at least.
+// Possible digest types supported by OcCryptoLib
+// for RSA verification
 //
-#if CONFIG_RSA_KEY_BIT_SIZE != 2048 || CONFIG_RSA_KEY_SIZE != 256
-#error "Only RSA-2048 is supported"
-#endif
+typedef enum RSA_DIGEST_TYPES_ {
+  RSA_DIGEST_TYPE_SHA256,
+  RSA_DIGEST_TYPE_SHA512,
+  RSA_DIGEST_TYPE_SHA384,
+} RSA_DIGEST_TYPES;
+
+typedef enum RSA_ALGORITHM_TYPES_ {
+  RSA_ALGORITHM_TYPE_NONE,
+  RSA_ALGORITHM_TYPE_SHA256_RSA2048,
+  RSA_ALGORITHM_TYPE_SHA256_RSA4096,
+  RSA_ALGORITHM_TYPE_SHA256_RSA8192,
+  RSA_ALGORITHM_TYPE_SHA512_RSA2048,
+  RSA_ALGORITHM_TYPE_SHA512_RSA4096,
+  RSA_ALGORITHM_TYPE_SHA512_RSA8192,
+  RSA_ALGORITHM_TYPE_SHA384_RSA2048,
+  RSA_ALGORITHM_TYPE_SHA384_RSA4096,
+  RSA_ALGORITHM_TYPE_SHA384_RSA8192,
+  RSA_ALGORITHM_NUM_TYPES
+} RSA_ALGORITHM_TYPES;
+
+//
+// Holds algorithm-specific data
+// The Padding is needed by RsaVerify
+//
+typedef struct RSA_ALGORITHM_DATA_ {
+  CONST UINT8  Padding;
+  UINTN        PaddingLen;
+  UINTN        HashLen;
+} RSA_ALGORITHM_DATA;
+
+#pragma pack(push, 1)
+
+/**
+  The header for a serialized RSA public key.
+   
+  Following this header is key_num_bits bits of N.
+  KeyNumBits of Rr. Both values are stored with most
+  significant bit first. Each serialized number takes up
+  KeyNumBits/8 bytes.
+
+ */
+typedef struct RSA_PUBLIC_KEY_HEADER_ {
+  UINT32  KeyNumBits;
+  UINT32  N0Inv;
+} RSA_PUBLIC_KEY_HEADER;
+
+#pragma pack(pop)
+
 
 #pragma pack(push, 1)
```

Comment:
Excuse me, I might be missing something here, but is RSA_PUBLIC_KEY not a type only used internally, i.e. either "hard-coded" (e.g. OcAppleKeysLib) or constructed at runtime from parsing? If so, why is it packed?